src/lib/services/pending-requests-interceptor.service.ts

```diff
@@ -33,35 +41,48 @@ export class PendingRequestsInterceptor implements HttpInterceptor {
     }
 
     get filteredUrlPatterns(): RegExp[] {
-        return this._filteredUrlPatterns;
+        return this._filters$.value.filteredUrlPatterns;
+    }
+
+    set filteredUrlPatterns(regexes: RegExp[]) {
+        this._filters$.next({
+            ...this._filters$.value,
+            filteredUrlPatterns: regexes
+        })
     }
 
     set filteredMethods(httpMethods: string[]) {
-        this._filteredMethods = httpMethods;
+        this._filters$.next({
+            ...this._filters$.value,
+            filteredMethods: httpMethods
+        })
     }
 
     set filteredHeaders(value: string[]) {
-        this._filteredHeaders = value;
+        this._filters$.next({
+            ...this._filters$.value,
+            filteredHeaders: value
+        })
     }
 
     set forceByPass(value: boolean) {
         this._forceByPass = value;
     }
 
     private shouldBypassUrl(url: string): boolean {
-        return this._filteredUrlPatterns.some(e => {
+        return this._filters$.value.filteredUrlPatterns.some(e => {
             return e.test(url);
         });
     }
 
-    private shouldBypassMethod(req: HttpRequest<unknown>): boolean {
-        return this._filteredMethods.some(e => {
+    private shouldBypassMethod(req: HttpRequest<any>): boolean {
+        return this._filters$.value.filteredMethods.some(e => {
             return e.toUpperCase() === req.method.toUpperCase();
         });
     }
 
-    private shouldBypassHeader(req: HttpRequest<unknown>): boolean {
-        return this._filteredHeaders.some(e => {
+    private shouldBypassHeader(req: HttpRequest<any>): boolean {
```

Comment:
rebase gone wrong

Reply:
Actually I think this is one of my edit...